src/common/components/messages/ProfileMessages.tsx

```diff
@@ -0,0 +1,385 @@
+import { useAnalytics } from '@/common/context/AnalyticsProvider';
+import { shortenAddress, showFirstAndLastFour } from '@/modules/utils/string';
+import { DateTime } from 'luxon';
+import { Button5 } from '../elements/Button2';
+import React, { FC, useState, useEffect, useMemo, useRef, Fragment, RefObject } from 'react';
+import * as web3 from '@solana/web3.js';
+import { Mailbox, MessageAccount } from '@usedispatch/client';
+import { ProfilePFP } from '@/common/components/feed/FeedCard';
+import classNames from 'classnames';
+import { useConnectedWalletProfile } from '@/common/context/ConnectedWalletProfileProvider';
+import { PencilAltIcon } from '@heroicons/react/outline';
+import { User } from '../feed/feed.utils';
+import { useGetProfilesQuery } from 'src/graphql/indexerTypes';
+import ProfileSearchCombobox from './ProfileSearchCombobox';
+
+interface ProfileMessagesInterface {
+  publicKey: string;
+  mailbox: Mailbox | undefined;
+  conversations: {
+    [conversationId: string]: MessageAccount[];
+  };
+  receiverAddress?: string;
+  addMessageToConversation: (conversationId: string, msg: MessageAccount) => void;
+}
+
+export const ProfileMessages = ({ publicKey, ...props }: ProfileMessagesInterface) => {
+  // Some notes for the future
+  // 1. This is more email than IM, from Dispatch's point of view anyway.
+  // 2. The messages listener can be used to notify people of new messages (as long as they stay on the site.). Should probably be put in the provider. Might be possible to integrate with dialect or notify in the future
+  // 3. TODO: Add in delete messages
+  const [newMessageText, setNewMessageText] = useState('');
+  const { mailbox, conversations, addMessageToConversation } = props;
+  const [recipient, setRecipient] = useState<User | null>(null);
+  const [messageTransactionInProgress, setmessageTransactionInProgress] = useState(false);
+
+  const contacts = Object.keys(conversations);
+
+  useEffect(() => {
+    if (props.receiverAddress && !recipient) {
+      setRecipient({
+        address: props.receiverAddress,
+      });
+    }
+  }, [props.receiverAddress]);
+
+  const { connectedProfile } = useConnectedWalletProfile();
+
+  const messagesInConversation: MessageAccount[] =
+    (recipient?.address && conversations[recipient.address]) || [];
+
+  const recipientInput = useRef<HTMLInputElement>(null);
+  const newMessageInput = useRef<HTMLTextAreaElement | HTMLInputElement>(null);
+
+  const { data } = useGetProfilesQuery({
+    variables: {
+      addresses: contacts,
+    },
+  });
+
+  const enrichedContacts = useMemo(
+    () =>
+      (
+        data?.wallets.slice() || // We need to do the sorting on a copy of the list returned from the query
```

Comment:
x ?? y vs x || y only makes a difference if x = 0 should return x instead of y, so it won't make a difference here

Reply:
We could add a "prefer ?? to ||" to the heuristics if you want though

Reply:
`??` is more specific than `||`, and I think we are better off with more specific when it can reduce bugs and make it easier to read code. I will add to the heuristics.